src/integrationtest/java/com/epam/idea/core/repository/IdeaRepositoryIntegTest.java

```diff
@@ -55,4 +55,22 @@ public void shouldFindAllIdeasByUserId() throws Exception {
 				.hasUsername("Jack")
 				.hasEmail("Jack@test.com");
 	}
+
+	@Test
+	@DatabaseSetup(value = "repository-idea-entries.xml")
+	public void shouldFindAllIdeasByTagId() throws Exception {
+		// Given:
+		long tagID = 1L;
+
+		// When:
+		List<Idea> ideas = ideaRepository.findByUserId(tagID);
```

Comment:
It must be `ideaRepository.findByTagId(tagID);`

Also I think u should change entities `id's` (so every entity will have unique id within file) in repository-idea-entries.xml (and all other tests xml files)  to avoid such failures in the future. Something like this:
```xml
    <user id="1" username="Jack" email="Jack@test.com" password="password" creation_time="2014-08-02 00:00:00"/>
    <idea id="2" title="Test title" description="Test description" creation_time="2014-08-03 00:00:00" modification_time="2014-09-04 00:00:00" rating="5" user_id="1"/>
    <tag id="3" name="Programming" />
    <idea_tag idea_id="2" tag_id="3"/>
```